### DISCUSSION TITLE
USB connection for sending data or serial communication

### DISCUSSION THREAD

**Discussion by DexterTaha** [pybricks/support#1628] [General]
I'm interested in exploring the possibility of using a USB connection for sending data or serial communication to run Python code on the Spike Prime Hub or the Robot Inventor Hub. I understand that this functionality might not be readily available, so I'm interested in any potential solutions or workarounds.

If there isn't an existing method to achieve this, could you provide some guidance or suggestions on how I might go about implementing a USB connection for this purpose? Any information or resources you can share would be greatly appreciated.

**Comment by dlech** [comment]
We have work-in-progress pull requests for this at https://github.com/pybricks/pybricks-micropython/pull/208 and https://github.com/pybricks/pybricksdev/pull/69.

If you can figure out why it makes the firmware crash, that would help move things along. 😄
